frontend/lib/src/WidgetStateManager.ts

```diff
@@ -680,9 +689,18 @@ export class WidgetStateManager {
   /**
    * Helper function to determine whether a form allows enter to submit
    * for input elements (st.number_input, st.text_input, etc.)
+   * First checks form's enterToSubmit param, otherwise default behavior:
    * Must be in a form & have 1st submit button enabled to allow
    */
-  public allowFormSubmitOnEnter(formId: string): boolean {
+  public allowFormEnterToSubmit(formId: string): boolean {
+    // Check user-set enterToSubmit param first (in FormState)
+    // Don't allow if false
+    const form = this.forms.get(formId)
+    if (notNullOrUndefined(form) && !form.enterToSubmit) {
```

Comment:
Can there be a case where `notNullOrUndefined(form)` is `false` (so form is `null`) and the rest of the function is still relevant? I would assume that, if the form is `false`, then there is also no submit button to check. So it might be a little bit cleaner to write:
```typescript
if (!notNullOrUndefined(form) || !form.enterToSubmit) {
  return false
}
```
to exit early when `form` is not set

Reply:
Definitely makes sense to early exit if not in form, thanks 👍🏼